### ISSUE TITLE
[suggestion] Add cli output json-formatter

### ISSUE THREAD

**Issue by lucaelin** [runem/lit-analyzer#254]
In order to improve integration into existing code-quality tools and tracker and ci-products like gitlab or sonar, a preferably eslint-compatible json output for the cli tool would be great.

**Comment by justinfagnani** [comment]
I think that the linting functionality of this tool should ultimately be moved to an eslint plugin, which would naturally make this integration possible. @rictic

**Comment by lucaelin** [comment]
I agree, but I also think that a machine-readable output format could still be useful and allow for more 3rd-party tooling to easily integrate with lit-analyzer, without depending on eslint.

**Comment by lucaelin** [comment]
I took the liberty to implement such a formatter, compatible with the codeclimate output. I don't mind this not being merged, because I didn't wait for any "go ahead", but it gave me a change to dig into the codebase and learn something new nonetheless :)
